src/ic_renderer.h

```diff
@@ -32,15 +38,19 @@ namespace IC {
         virtual void AddDirectionalLight(std::shared_ptr<DirectionalLight> light) = 0;
         virtual void DrawFrame() = 0;
 
-        void AddImguiFunction(std::function<void()> function);
-        void RemoveImguiFunction(std::function<void()> function);
+        void AddImguiFunction(std::string windowName, std::function<void()> function);
+        void RemoveImguiFunction(std::string windowName);
 
     protected:
-        std::vector<std::function<void()>> imGuiFunctions;
+        std::unordered_map<std::string, std::function<void()>> imGuiFunctions;
         GLFWwindow *window;
+        RenderStats renderStats{};
+
+        void RenderStatsGUI();
 
     private:
         Renderer(const Renderer &) = delete;
         Renderer &operator=(const Renderer &) = delete;
+        const std::string STATS_WINDOW_NAME = "render stats";
```

Comment:
Can you make this a `static constexpr`? I don't remember if that's allowed for `std::string`s but if not can you at least make it `static const`?